include/boost/url/any_allocator.hpp

```diff
@@ -0,0 +1,165 @@
+//
+// Copyright (c) 2022 Alan Freitas (alandefreitas@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/CPPAlliance/url
+//
+
+#ifndef BOOST_URL_ANY_ALLOCATOR_HPP
+#define BOOST_URL_ANY_ALLOCATOR_HPP
+
+#include <boost/url/detail/config.hpp>
+
+#include <memory>
+#include <cstddef>
+#include <type_traits>
+#include <atomic>
+
+namespace boost {
+namespace urls {
+
+// move to another header
+namespace detail {
+
+using type_id_t = void const*;
+
+template<class T>
+type_id_t
+get_type_id() noexcept
+{
+    static constexpr char c{};
+    return &c;
+};
+
+struct any_allocator_base {
```

Comment:
braces need to go on their own line or else it makes debugging more difficult !!!